---
bool success
string message
string[] data
